google/utils.go

```diff
@@ -251,8 +251,9 @@ func isConflictError(err error) bool {
 }
 
 func linkDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
-	parts := strings.Split(old, "/")
-	if parts[len(parts)-1] == new {
+	oldParts := strings.Split(old, "/")
```

Comment:
The problem with updating this method is that this will cause issue for zonal or regional self-link

If field was "regions/us-east1/someResource/myResource" and now is "regions/us-west1/someResource/myResource" this would suppress the diff. 

I wouldn't update this method and use the other more robust `compareSelfLinkOrResourceName`